### PR TITLE
fix: added configurable padding

### PR DESCRIPTION
**Pull request by vanlooverenkoen**
## Description

Add the option to use padding around a GoldenTestScenario

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by btrautmann — COMMENTED** [review]
Is this purely used for aesthetics?

**Comment by vanlooverenkoen** [comment]
Because we have a lot of GoldenTestScenarios with the removal of the padding widget it is a lot harder to do visual pr review. The cognitive load is much higher when there is no padding. (I get that this is not required for everyone, that is why I made it optionally configurable)

**Comment by btrautmann** [comment]
> Because we have a lot of GoldenTestScenarios with the removal of the padding widget it is a lot harder to do visual pr review. The cognitive load is much higher when there is no padding. (I get that this is not required for everyone, that is why I made it optionally configurable)

Understood. Yeah I think `GoldenTestTheme` created a nice place to put these optional parameters. I'm cool with this 👍

**Review by btrautmann — COMMENTED** [review]
Can we just add a test for this?

**Comment by vanlooverenkoen** [comment]
Test added ✅

File: test/src/golden_test_adapter_test.dart [hunk 1]
```diff
@@ -316,6 +317,55 @@ void main() {
         expect(tester.platformDispatcher.textScaleFactor, 2.0);
       });
 
+      group('padding', () {
```
Comment (jeroen-meijer) [inline test/src/golden_test_adapter_test.dart:320]: _(Bit nit picky, and these tests are nice, but in the name of testing things properly:)_

This test group tests two assumptions at once. We should ideally split them up like so:

1. In this file, we should only check that the `padding` provided to the `GoldenTestTheme` (whatever padding value that may be) is actually applied in the correct place in the rendered golden file (much like you currently do on line `337`, though arguably that doesn't check whether the padding is applied around the correct widget, only that *any* padding widget exists with that value).

2. In a _separate_ test in `golden_test_theme_test.dart` we should check whether the default `padding` value is set correctly.

File: lib/src/golden_test_theme.dart [hunk 2]
```diff
@@ -30,14 +32,19 @@ class GoldenTestTheme extends ThemeExtension<GoldenTestTheme> {
   /// The border color used to separate scenarios in a [GoldenTestGroup].
   final Color borderColor;
 
+  /// The padding that is used to wrap a [GoldenTestScenario]
+  final EdgeInsetsGeometry padding;
```
[the remark this answers is not in the document]
Comment (mx1up) [reply lib/src/golden_test_theme.dart:36]: yes, please apply the same padding in `GoldenTestScenario` (which was also removed in 0.10.0), or provide a separate padding configuration for the scenario and for the individual adapter.

**Comment by btrautmann** [comment]
> > Is this purely used for aesthetics?
> 
> it seems in 0.10.0, a breaking change was introduced that removed the padding ( [5139917](https://github.com/Betterment/alchemist/commit/5139917e4064811c2dadc153ffcf40f07f102459) ). But since it is not documented, maybe it was an accidental change? If so, it might be better to introduce a default padding of 8dp as it used to be before.

The change was intentional, I'd rather that padding be configurable than to add it back.

**Comment by mx1up** [comment]
> > it seems in 0.10.0, a breaking change was introduced that removed the padding ( [5139917](https://github.com/Betterment/alchemist/commit/5139917e4064811c2dadc153ffcf40f07f102459) ). But since it is not documented, maybe it was an accidental change? If so, it might be better to introduce a default padding of 8dp as it used to be before.
> 
> The change was intentional, I'd rather that padding be configurable than to add it back.

ok, fine. just a suggestion: it might be helpful to document this change somewhere for users migrating to 0.10+

**Comment by btrautmann** [comment]
> ok, fine. just a suggestion: it might be helpful to document this change somewhere for users migrating to 0.10+

Yeah, apologies for not being more transparent in the changelog about the change. Once this lands (CC @vanlooverenkoen) I'll make sure to make note of it in the changelog indicating the correct path for migrating.

**Comment by vanlooverenkoen** [comment]
@btrautmann  fixed all comments

**Comment by mx1up** [comment]
> @btrautmann fixed all comments

thanks @vanlooverenkoen , i can confirm, using this branch, all our golden tests are green again 🥳

**Comment by vanlooverenkoen** [comment]
Nice!! Do you like the documentation I added?

**Comment by mx1up** [comment]
> Nice!! Do you like the documentation I added?

yes, clear to me 👌 

I don't know if this project follows Effective dart doc recommendations though, specifically:

[DO start doc comments with a single-sentence summary](https://dart.dev/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary)

**Comment by btrautmann** [comment]
@vanlooverenkoen thanks for updates; unfortunately it looks like we have a couple failing tests.

**Comment by mx1up** [comment]
@vanlooverenkoen any progress on this? 🙏

**Comment by vanlooverenkoen** [comment]
@mx1up & @btrautmann should be fixed (at least locally)

**Review by btrautmann — APPROVED** [review]
domain lgtm 🚀 
platform lgtm 💪

**Pull request merged**